wallet/wallet.c

```diff
@@ -799,43 +776,16 @@ static bool wallet_ever_used(struct wallet *w)
  */
 u32 wallet_first_blocknum(struct wallet *w, u32 first_possible)
 {
-	int err;
-	u32 first_channel, first_utxo;
-	sqlite3_stmt *stmt =
-	    db_query(__func__, w->db,
-		     "SELECT MIN(first_blocknum) FROM channels;");
-
-	/* If we ever opened a channel, this will give us the first block. */
-	err = sqlite3_step(stmt);
-	if (err == SQLITE_ROW && sqlite3_column_type(stmt, 0) != SQLITE_NULL)
-		first_channel = sqlite3_column_int(stmt, 0);
-	else
-		first_channel = UINT32_MAX;
-	sqlite3_finalize(stmt);
-
-#ifdef COMPAT_V052
-	/* This field was missing in older databases. */
-	first_utxo = db_get_intvar(w->db, "last_processed_block", 0);
-	if (first_utxo == 0) {
-		/* Don't know?  New db, or upgraded. */
-		if (wallet_ever_used(w))
-			/* Be conservative */
-			first_utxo = first_possible;
-		else
-			first_utxo = UINT32_MAX;
-	}
-#else
-	first_utxo = db_get_intvar(w->db, "last_processed_block", UINT32_MAX);
-#endif
-
-	/* Never go below the start of the Lightning Network */
-	if (first_utxo < first_possible)
-		first_utxo = first_possible;
+	u32 blockheight;
+	sqlite3_stmt *stmt = db_prepare(w->db, "SELECT MAX(height) FROM blocks;");
 
-	if (first_utxo < first_channel)
-		return first_utxo;
-	else
-		return first_channel;
+	/* If we ever processed a block we'll get the latest block in the chain */
+	if (sqlite3_step(stmt) == SQLITE_ROW && sqlite3_column_type(stmt, 0) != SQLITE_NULL) {
+		blockheight = sqlite3_column_int(stmt, 0);
+		sqlite3_finalize(stmt);
+		return blockheight;
+	} else
+		return first_possible;
 }
```

Comment:
If they are currently onchain, and upgrade, this will break them?  I think this logic needs to stay for a while :(